### PR TITLE
support configurable compress level

### PR DESCRIPTION
**Pull request by SimbaGithub**
Fix for simba ticket 00316238. Make compress level configurable. Will make another PR on ODBC side when this PR is merged.

### CODE REVIEW

**Review by sfc-gh-cshi — COMMENTED** [review]
I think this is pretty much good to go. But I think since we introduce a configuration here, who will be responsible to make the correspondent document changes?

File: cpp/util/CompressionUtil.cpp [hunk 1]
```diff
@@ -37,7 +38,11 @@ int Snowflake::Client::Util::CompressionUtil::compressWithGzip(FILE *source,
   strm.zalloc = Z_NULL;
   strm.zfree = Z_NULL;
   strm.opaque = Z_NULL;
-  ret = deflateInit2(&strm, Z_DEFAULT_COMPRESSION, Z_DEFLATED,
+  if ((level < 0) || (level > 9))
+  {
+      level = Z_DEFAULT_COMPRESSION;
```
[the remark this answers is not in the document]
Comment (SimbaGithub) [reply cpp/util/CompressionUtil.cpp:43]: > so when level<0, we assign it Z_DEFAULT_COMPRESSION, I think it is -1 and still less than 0?

Yes, Z_DEFAULT_COMPRESSION is the only valid value that less than 0.

**Comment by SimbaGithub** [comment]
> I think this is pretty much good to go. But I think since we introduce a configuration here, who will be responsible to make the correspondent document changes?

Not sure if we have documentation for libsnowflakeclient. When this PR is merged, will make another PR on odbc side, and the new setting will be documented on ODBC side.

**Review by sfc-gh-cshi — APPROVED** [review]
LGTM

File: cpp/util/CompressionUtil.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfc-gh-sleslie) [reply cpp/util/CompressionUtil.cpp:43]: Hello @sfc-gh-cshi ! My name is Steven. I just joined the Snowflake documentation team last week. I'm working on documenting this change this week.

[the remark this answers is not in the document]
Comment (SimbaGithub) [reply cpp/util/CompressionUtil.cpp:43]: Hi @sfc-gh-sleslie, FYI, the spec of this feature is noted in this PR https://github.com/snowflakedb/snowflake-odbc/pull/278 on ODBC side.

[the remark this answers is not in the document]
Comment (sfc-gh-sleslie) [reply cpp/util/CompressionUtil.cpp:43]: Thanks @SimbaGithub ! I saw that PR earlier. I have some questions that I will post soon in the simba-snowflake channel.

**Pull request merged**